### PR TITLE
chore(psycopg2): update integration implementation

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

The earlier supported version of `psycopg2` is 2.7, which means that the checks for this version can be removed. Moreover, the `parse_dsn` method is already available on this version so we can use that instead of a custom implementation.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/contrib/psycopg/patch.py [hunk 1]
```diff
@@ -47,14 +45,15 @@ def unpatch():
     if getattr(psycopg2, "_datadog_patch", False):
         setattr(psycopg2, "_datadog_patch", False)
         psycopg2.connect = _connect
+        _unpatch_extensions(_psycopg2_extensions)
```
Comment (brettlangdon) [inline ddtrace/contrib/psycopg/patch.py:48]: is this a bug?

Comment (P403n1x87) [reply ddtrace/contrib/psycopg/patch.py:48]: I thought it was weird it was not used here, but maybe there is a reason for not unpatching?

File: ddtrace/contrib/sqlalchemy/engine.py [hunk 2]
```diff
@@ -27,6 +27,12 @@
 from ...pin import Pin
 
 
+try:
+    from psycopg2.extensions import parse_dsn
+except ImportError:
+    parse_dsn = None
```
[the remark this answers is not in the document]
Comment (P403n1x87) [reply ddtrace/contrib/sqlalchemy/engine.py:33]: Yes the comment suggested using the psycopg2 implementation once it got merged.

File: ddtrace/contrib/sqlalchemy/engine.py [hunk 3]
```diff
@@ -144,10 +150,11 @@ def _set_tags_from_url(span, url):
...
     if "postgres" == vendor:
-        if hasattr(cursor, "connection") and hasattr(cursor.connection, "dsn"):
+        if hasattr(cursor, "connection"):
             dsn = getattr(cursor.connection, "dsn", None)
             if dsn:
-                d = sqlx.parse_pg_dsn(dsn)
-                span.set_tag(sqlx.DB, d.get("dbname"))
-                span.set_tag(netx.TARGET_HOST, d.get("host"))
-                span.set_tag(netx.TARGET_PORT, d.get("port"))
+                assert parse_dsn is not None, "psycopg2 must be available"
```
[the remark this answers is not in the document]
Comment (P403n1x87) [reply ddtrace/contrib/sqlalchemy/engine.py:156]: Yeah this wouldn't work with another library. Do we use our own implementation as a fallback?

[the remark this answers is not in the document]
Comment (brettlangdon) [reply ddtrace/contrib/sqlalchemy/engine.py:156]: Not sure... it is probably the safer way to handle it

98% of the time it'll use the psycopg2 version, and in the off chance someone is using a different engine, then we'll still support it as best we can

File: ddtrace/ext/sql.py [hunk 4]
```diff
@@ -16,13 +20,15 @@ def normalize_vendor(vendor):
         return vendor
 
 
-def parse_pg_dsn(dsn):
-    """
-    Return a dictionary of the components of a postgres DSN.
+try:
+    from psycopg2.extensions import parse_dsn as parse_pg_dsn  # type: ignore[import]
+except ImportError:
```
Comment (brettlangdon) [inline ddtrace/ext/sql.py:25]: 👍🏻

**Review by brettlangdon — APPROVED** [review]
regression test would be nice, but otherwise lgtm

File: ddtrace/contrib/psycopg/patch.py (hunk 1 above)
Comment (P403n1x87) [reply ddtrace/contrib/psycopg/patch.py:48]: We have this already, so we should be good?

https://github.com/DataDog/dd-trace-py/blob/7f11feb46abb928c69bdd566a0632736ee875119/tests/contrib/psycopg/test_psycopg.py#L55-L79

Comment (brettlangdon) [reply ddtrace/contrib/psycopg/patch.py:48]: but why wasn't it failing before?

Comment (P403n1x87) [reply ddtrace/contrib/psycopg/patch.py:48]: hmm ... my guess is that this test is insensitive to the extensions being patched/unpatched

**Review by jd — APPROVED** [review]

**Pull request merged**